File: Non-overlapping Intervals/non-overlapping-intervals.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end());
        int n = intervals.size();
        int i = 0, j = 1;
        int count = 0;
        while (j < n) {
            int s = intervals[i][0];
            int e = intervals[i][1];
            int ns = intervals[j][0];
            int ne = intervals[j][1];
            if (e <= ns) {
                i=j;
                j++;
            } else {
                if (e >= ne) {
                    i = j;
                }
                j++;
                count++;
            }
        }
        return count;
    }
};

File: Non-overlapping Intervals/non-overlapping-intervals.py
class Solution(object):
    def eraseOverlapIntervals(self, intervals):
        """
        :type intervals: List[List[int]]
        :rtype: int
        """
        intervals.sort(key=lambda x: x[0])
        n = len(intervals)
        i, j = 0, 1
        count = 0
        
        while j < n:
            s, e = intervals[i]
            ns, ne = intervals[j]
            
            if e <= ns:
                i = j
                j += 1
            else:
                if e >= ne:
                    i = j
                j += 1
                count += 1
        
        return count
